internal/utils/comparators/deployment/comparator.go

```diff
@@ -0,0 +1,80 @@
+package deployment
+
+import (
+	"reflect"
+
+	log "github.com/ViaQ/logerr/v2/log/static"
+	"github.com/openshift/cluster-logging-operator/internal/utils"
+	apps "k8s.io/api/apps/v1"
+)
+
+// AreSame compares deployments for equality and return true equal otherwise false
+func AreSame(current *apps.Deployment, desired *apps.Deployment) (bool, string) {
```

Comment:
There is a lot of duplication in the meat of this function.  I believe we mostly are diffing the podspec or related template.  Can we refactor into maybe a pod comparitor or something

Reply:
These comparators are only comparing PodSpecs so I refactored into `pod/comparator.go` I removed deployment and daemonset comparators.